code/datums/components/food/edible.dm

```diff
@@ -264,7 +264,8 @@ Behavior that's still missing from this component that original food items had t
 
 	if(feeder.a_intent == INTENT_HARM)
 		return
-	if(!owner.reagents.total_volume)//Shouldn't be needed but it checks to see if it has anything left in it.
+	//If we had initial volume we should still have some until the last bite
+	if(volume > 0 && !owner.reagents.total_volume)
```

Comment:
This makes no sense, if volume == 0 then there shouldnt be an edible component.

Reply:
you are assume that you are the only one that can touch reagents, this object self manages reagents and needs a flag for that in your component, a simple solution was to tell the component that it if volume was set to 0 then do not assume reagent control.
I can add a flag for that to the edible component if you like.

Reply:
You shouldn't be adding edible components with 0 volume, period. Adding a flag is not the right solution for this.

The problem isnt the edible component, its this broken-ass stomach change.